### PR TITLE
Improve typing notification animations

### PR DESCRIPTION
**Pull request by jmartinesp**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : UX improvement

## Content

- Make typing notifications appear using an animation, and keep the extra bottom space after the first one is displayed.
- Also, simplify how `MigrationScreenView` animation works, since I forgot `AnimatedVisibility()` existed.

## Motivation and context

- Iterate on the initial implementation.

## Screenshots / GIFs

https://github.com/element-hq/element-x-android/assets/480955/a216443d-75cb-4fe6-9710-a4c075cfcc21

## Tests

Open a room, scroll to the bottom, wait until you receive typing notifications. You should see:

- An extra space is added at the bottom once the first typing notification appears.
- The typing notifications will appear and disappear using a fade in/out animation.
- The extra space is still there after the first notification disappears, so we only resize it once.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.typing_TypingNotificationView_null_TypingNotificationView-Day-60_60_null_8,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/ui_S_t[f.messages.impl.typing_TypingNotificationView_null_TypingNotificationView-Day-60_60_null_8,NEXUS_5,1.0,en].png:1]: The screenshot is not very representative 😅 . There should be an empty space at the bottom, I'll try to make it visible against the theme background.

**Review by bmarty — COMMENTED** [review]
Thanks, please see my remarks, and let me know if you have any question

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/typing/TypingNotificationPresenter.kt [hunk 1]
```diff
@@ -54,9 +56,18 @@ class TypingNotificationPresenter @Inject constructor(
             }
         }
 
+        // This will keep the space reserved for the typing notifications after the first one is displayed
+        var reserveSpace by remember { mutableStateOf(false) }
+        LaunchedEffect(renderTypingNotifications, typingMembersState.value) {
+            if (renderTypingNotifications && typingMembersState.value.isNotEmpty()) {
+                reserveSpace = true
```
[the remark this answers is not in the document]
Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/typing/TypingNotificationPresenter.kt:63]: After discussing with @stefanceriu (who discussed with @amshakal) this is maybe not necesssary, but having an empty space at the bottom of the timeline is not great to me. Not a blocker so.

[the remark this answers is not in the document]
Comment (jmartinesp) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/typing/TypingNotificationPresenter.kt:63]: I just spoke with @stefanceriu and he confirmed we've implemented the same behaviour in both platforms: the extra space should stay until you exit the room.

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**